stl/inc/xmemory

```diff
@@ -157,34 +172,46 @@ inline void _Adjust_manually_vector_aligned(void*& _Ptr, size_t& _Bytes) {
 #ifdef __cpp_aligned_new
 template <size_t _Align, class _Traits = _Default_allocate_traits,
     enable_if_t<(_Align > __STDCPP_DEFAULT_NEW_ALIGNMENT__), int> = 0>
-__declspec(allocator) void* _Allocate(const size_t _Bytes) {
+_CONSTEXPR20_DYNALLOC __declspec(allocator) void* _Allocate(const size_t _Bytes) {
     // allocate _Bytes when __cpp_aligned_new && _Align > __STDCPP_DEFAULT_NEW_ALIGNMENT__
     if (_Bytes == 0) {
         return nullptr;
     }
 
-    size_t _Passed_align = _Align;
+#ifdef __cpp_lib_is_constant_evaluated
+    if (_STD is_constant_evaluated()) {
+        return _Traits::_Allocate(_Bytes);
+    } else
+#endif // __cpp_lib_is_constant_evaluated
+    {
+        size_t _Passed_align = _Align;
 #if defined(_M_IX86) || defined(_M_X64)
-    if (_Bytes >= _Big_allocation_threshold) {
-        // boost the alignment of big allocations to help autovectorization
-        _Passed_align = (_STD max)(_Align, _Big_allocation_alignment);
-    }
+        if (_Bytes >= _Big_allocation_threshold) {
+            // boost the alignment of big allocations to help autovectorization
+            _Passed_align = (_STD max)(_Align, _Big_allocation_alignment);
+        }
 #endif // defined(_M_IX86) || defined(_M_X64)
-
-    return _Traits::_Allocate_aligned(_Bytes, _Passed_align);
+        return _Traits::_Allocate_aligned(_Bytes, _Passed_align);
+    }
 }
 
 template <size_t _Align, enable_if_t<(_Align > __STDCPP_DEFAULT_NEW_ALIGNMENT__), int> = 0>
-void _Deallocate(void* _Ptr, const size_t _Bytes) noexcept {
+_CONSTEXPR20_DYNALLOC void _Deallocate(void* _Ptr, const size_t _Bytes) noexcept {
     // deallocate storage allocated by _Allocate when __cpp_aligned_new && _Align > __STDCPP_DEFAULT_NEW_ALIGNMENT__
-    size_t _Passed_align = _Align;
+#ifdef __cpp_lib_is_constant_evaluated
+    if (_STD is_constant_evaluated()) {
+        ::operator delete(_Ptr);

```

Comment:
I am wondering whether this should be `_Traits::_Deallocate(_Ptr);`

Reply:
Or above we should call `::operator new` in `_Allocate`

Reply:
This change is no longer present in `<xmemory>`.